WalletWasabi/Helpers/Startup/MacOsStartupChecker.cs

```diff
@@ -0,0 +1,37 @@
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Microservices;
+
+namespace WalletWasabi.Helpers
+{
+	public static class MacOsStartupChecker
+	{
+		private const string ListCmd = "osascript -e \' tell application \"System Events\" to get every login item\'";
+
+		public static async Task<bool> CheckLoginItemExistsAsync()
+		{
+			var escapedArgs = ListCmd.Replace("\"", "\\\"");
+
+			var startInfo = new ProcessStartInfo
+			{
+				FileName = "/usr/bin/env",
+				Arguments = $"sh -c \"{escapedArgs}\"",
+				RedirectStandardOutput = true,
+				UseShellExecute = false,
+				CreateNoWindow = true,
+				WindowStyle = ProcessWindowStyle.Hidden
+			};
+
+			using var process = new ProcessAsync(startInfo);
+
+			process.Start();
+
+			string output = await process.StandardOutput.ReadToEndAsync().ConfigureAwait(false);
+
+			await process.WaitForExitAsync(CancellationToken.None).ConfigureAwait(false);
```

Comment:
One risk of not having a cancellation token here is that you might wait indefinitely if there is a change in how `osascript` behaves (a use confirmation (Y/N) on `osascript` side comes to mind).

It may make sense to propagate a cancellation token here. An alternative might be to create a new token and give it some time budget (like 15 seconds or something) to finish.

Reply:
Should I send down a `CancellationToken` from `Global.cs` or is it enough to create it locally, with a timespan (10-15 sec) and dispose it after?

Reply:
Best to ask @molnard about this not to waste your time.

Reply:
I would add a Timeout here: https://github.com/molnard/WalletWasabi/blob/9010473a2c43dc150aa377b9e0193e7afffd3385/WalletWasabi.Gui/Global.cs#L252

And add GetCurrentValueAsync(CancellationToken cancel = default)